source/aws_signing.c

```diff
@@ -577,6 +578,18 @@ static int s_add_authorization_query_params(struct aws_signing_state_aws *state,
         goto done;
     }
 
+    /* X-Amz-Security-token */
+    if (state->config->credentials->session_token) {
```

Comment:
This adds the param to what gets signed.  Does this need to be configurable (aws_signing_config boolean), defaulting to true?